tests/fastify-plugins/maps.js

```diff
@@ -123,3 +204,102 @@ function setup(t) {
 
   return server
 }
+
+function setupFetchMock() {
```

Comment:
Probably not worth it, but I could see creating this `MockAgent` and passing it into the `fetch` using [the `dispatcher` option](https://undici.nodejs.org/#/?id=undicifetchinput-init-promise). I think what you have here is fine.

Reply:
useful to know that, but I don't think it would handle the case where the code we're testing is calling fetch internally. the global dispatcher approach makes sure that we can intercept those underlying fetch calls